ambari-server/src/main/python/ambari_server/setupTrustedProxy.py

```diff
@@ -0,0 +1,154 @@
+#!/usr/bin/env python
+
+'''
+Licensed to the Apache Software Foundation (ASF) under one
+or more contributor license agreements.  See the NOTICE file
+distributed with this work for additional information
+regarding copyright ownership.  The ASF licenses this file
+to you under the Apache License, Version 2.0 (the
+"License"); you may not use this file except in compliance
+with the License.  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+'''
+
+import re
+import urllib2
+
+from ambari_commons.exceptions import FatalException, NonFatalException
+from ambari_commons.logging_utils import get_silent, print_info_msg
+from ambari_server.serverConfiguration import get_ambari_properties
+from ambari_server.serverUtils import is_server_runing, get_ambari_admin_username_password_pair, \
+  get_cluster_name, perform_changes_via_rest_api, get_json_via_rest_api, get_value_from_dictionary
+from ambari_server.setupSecurity import REGEX_TRUE_FALSE
+from ambari_server.userInput import get_validated_string_input, get_YN_input
+
+
+PROXYUSER_PLACEHOLDER = "$PROXY_USER"
+TPROXY_SUPPORT_ENABLED = "ambari.tproxy.authentication.enabled"
+PROXYUSER_HOSTS = "ambari.tproxy.proxyuser." + PROXYUSER_PLACEHOLDER + ".hosts"
+PROXYUSER_USERS = "ambari.tproxy.proxyuser." + PROXYUSER_PLACEHOLDER + ".users"
+PROXYUSER_GROUPS = "ambari.tproxy.proxyuser." + PROXYUSER_PLACEHOLDER + ".groups"
+
+TPROXY_CONFIG_API_ENTRYPOINT = 'services/AMBARI/components/AMBARI_SERVER/configurations/tproxy-configuration'
+
+REGEX_ANYTHING = ".*"
+WILDCARD_FOR_ALL = "*"
+
+
+def get_trusted_proxy_properties(ambari_properties, admin_login, admin_password):
+  print_info_msg("Fetching Trusted Proxy configuration from DB")
+
+  try:
+    response_code, json_data = get_json_via_rest_api(ambari_properties, admin_login, admin_password, TPROXY_CONFIG_API_ENTRYPOINT)
+  except urllib2.HTTPError as http_error:
+    if http_error.code == 404:
+      # This means that there is no Trusted Proxy configuration in the database yet -> we can not fetch the property (but this is NOT an error)
+      json_data = None
+    else:
+      raise http_error
+
+  if json_data and 'Configuration' in json_data and 'properties' in json_data['Configuration']:
```

Comment:
Consider using a_dict.get(key, default) instead of the _in_ checks 

For example: 

```python
json_data.get('Configuration', {}).get('properties', {})
```